test/functional/admin/legacy_responses_controller_test.rb

```diff
@@ -0,0 +1,116 @@
+require "test_helper"
+
+class Admin::LegacyResponsesControllerTest < ActionController::TestCase
+  tests Admin::ResponsesController
+
+  legacy_should_be_an_admin_controller
+
+  setup do
+    login_as :writer
```

Comment:
Thanks for duplicating the controller! I think we just need to update the non-legacy controller to use the `login_as_preview_design_system_user` so it runs that set of tests with a user who has the permission

Reply:
@davidgisbey I've updated the test here to use the appropriate login and removed the legacy one previously added. I'm not entirely confident that's what's required here though. See the [last commit](https://github.com/alphagov/whitehall/pull/7369/commits/7c7f8f73ab33c21cf57fc85109ca8a1f237b14a0). Could you check that?